GitUI/UserControls/RevisionGrid/Graph/RevisionGraph.cs

```diff
@@ -212,7 +213,7 @@ public void Add(GitRevision revision, RevisionNodeFlags types)
             }
 
             // Ensure all parents are loaded before adding it to the _nodes list. This is important for ordering.
-            _nodes.Add(revisionGraphRevision);
+            ImmutableInterlocked.Update(ref _nodes, list => list.Add(revisionGraphRevision));
```

Comment:
The adding is always done in the same thread. It is safe to assume that list.add is always called from the same thread. I think we do not need to use ImmutableInterlocked here.